sda/cmd/api/api.go

```diff
@@ -317,39 +317,73 @@ func getFiles(c *gin.Context) {
 	c.JSON(200, files)
 }
 
+/*
+ingestFile handles requests to initiate ingestion of a file.
+This endpoint supports two input modes:
+1. By file ID (via the "fileid" query parameter): Looks up the user and file path from the database.
+2. By JSON payload: Expects a JSON body with user and file path.
+The function constructs an ingest message, validates it
+and sends it to the broker with the appropriate correlation ID.
+*/
 func ingestFile(c *gin.Context) {
-	var ingest schema.IngestionTrigger
-	if err := c.BindJSON(&ingest); err != nil {
-		c.AbortWithStatusJSON(
-			http.StatusBadRequest,
-			gin.H{
-				"error":  "json decoding : " + err.Error(),
-				"status": http.StatusBadRequest,
-			},
-		)
+	var (
+		ingest schema.IngestionTrigger
+		corrID string
+	)
+	switch {
+	case c.Query("fileid") != "" && c.Request.ContentLength > 0:
+		c.AbortWithStatusJSON(http.StatusBadRequest, "both file ID parameter and payload provided.")
```

Comment:
```suggestion
```

We shouldn't need to check this. If a file id is given then we are happy since we can get all we require to ingest the file.

Reply:
I think this check might be useful, especially when people call the API manually.